### PR TITLE
Add a stack.yaml to get IDE code completion

### PR DESCRIPTION
**Pull request by arichardson**
This gives me inline documentation hover and warnings, so I might be able to add a v2 trace format without really understanding haskell.
Screenshot 2020-11-17 at 12 00 43

### CODE REVIEW

**Comment by PeterRugg** [comment]
Any reason for the change in QuickCheck pin?

File: stack.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+#   - auto-update
+#   - wai
+packages:
+- .
+# Dependency packages to be pulled from upstream that are not in the resolver.
+# These entries can reference officially published versions as well as
+# forks / in-progress versions pinned to a git hash. For example:
+#
+extra-deps:
+- bitwise-1.0.0.1
```
Comment (jrtc27) [inline stack.yaml:38]: Why's this needed?

**Comment by jrtc27** [comment]
I think all (or at least most of) the boilerplate comments in the file should just be dropped.

File: stack.yaml (hunk 1 above)
Comment (arichardson) [reply stack.yaml:38]: Apparently bitwise is not part of the packages that are included in the stack repo.

**Comment by arichardson** [comment]
> Any reason for the change in QuickCheck pin?

It appears like version 2.12 does exist in the stack repos, there is only that version.

The other option that it suggested to me was to add 
`- QuickCheck-2.12@sha256:b59268ced7aae700be5a90251ff8df017b5fa4d4625cc3a966be9668b9fe574b,6328` to `extra-deps`

File: QCVEngine.cabal [hunk 2]
```diff
@@ -54,5 +54,5 @@ executable QCVEngine
   build-depends:       base >=4.9, time, split >=0.2, regex-posix >=0.95,
                        binary >=0.8, bitwise >=1.0, network >=2.8, parsec,
                        bytestring >=0.10, filemanip >=0.3,
-                       QuickCheck ==2.12
+                       QuickCheck ==2.12.6.1
```
Comment (jrtc27) [inline QCVEngine.cabal:57]:
```suggestion
                       QuickCheck >=2.12, QuickCheck <2.13
```
then?

Comment (arichardson) [reply QCVEngine.cabal]: Seems like the syntax is `QuickCheck >= 2.12 && < 2.13`. Thanks.

**Pull request merged**